src/core/lib/stripe.ts

```diff
@@ -1,7 +1,185 @@
 import Stripe from "stripe";
 import config from "@config";
+import OptionsService from "@core/services/OptionsService";
+import currentLocale, { locales } from "@locale";
 
-export default new Stripe(config.stripe.secretKey, {
+export const stripe = new Stripe(config.stripe.secretKey, {
   apiVersion: "2023-10-16",
   typescript: true
 });
+
+/**
+ * Get the tax rates by display name
+ * @param displayName The display name
+ * @param params The stripe request options, e.g. `{ limit: 1, active: true }`
+ * @returns The tax rates
+ */
+export const stripeTaxRatesGetByDisplayName = async (
+  displayName: string,
+  params?: Stripe.TaxRateListParams
+) => {
+  return (await stripe.taxRates.list(params)).data.filter(
+    (taxRate) => taxRate.display_name === displayName
+  );
+};
+
+/**
+ * Get the default display name for the tax rate.
+ *
+ * @returns The default display name
+ */
+export const stripeTaxRateGetDefaultDisplayName = () =>
+  (currentLocale().membershipBuilder.steps.joinForm as any).taxRate ||
+  locales.en.membershipBuilder.steps.joinForm.taxRate;
+
+/**
+ * Get the default tax rate.
+ * This method gets the tax rate directly from stripe, so no request to our own database is made.
+ *
+ * @param active The active status of the tax rate, keep undefined to get all tax rates, also the disabled ones
+ * @returns The default tax rate
+ */
+export const stripeTaxRateGetDefault = async (
+  active?: boolean
+): Promise<Stripe.TaxRate | undefined> => {
+  const defaultId = OptionsService.get("tax-rate-stripe-default-id").value;
+  if (defaultId) {
+    const taxRate = await stripe.taxRates.retrieve(defaultId);
+    if (taxRate.active === active) {
+      return taxRate;
+    }
+  }
+  const params: Stripe.TaxRateListParams = { limit: 1 };
+  if (active) {
+    params.active = active;
+  }
+  return (
+    await stripeTaxRatesGetByDisplayName(
+      stripeTaxRateGetDefaultDisplayName(),
+      params
+    )
+  )?.[0];
+};
+
+/**
+ * Rename the tax rate.
+ * When existing tax rates are archived, we rename them with this function.
+ * @param displayName
+ * @param percentage
+ * @returns
+ */
+const renameTaxRate = (displayName: string, percentage: number) => {
```

Comment:
Maybe `getTaxRateName` is slightly clearer?

Reply:
yes, thanks